pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -500,6 +510,34 @@ func (gce *GCECloud) ensureTargetPoolAndHealthCheck(tpExists, tpNeedsRecreation
 			glog.Infof("ensureTargetPoolAndHealthCheck(%s): Updated target pool (with %d hosts).", lbRefStr, len(hosts)-maxTargetPoolCreateInstances)
 		}
 	} else if tpExists {
+
+		if hcToDelete != nil {
+			if hc, err := gce.GetHttpHealthCheck(hcToDelete.Name); err != nil {
+				if !isHTTPErrorCode(err, http.StatusNotFound) {
+					return fmt.Errorf("error getting HTTP health check for load balancer (%s): %v", lbRefStr, err)
+				}
+			} else {
+				healthchecks := []*compute.HealthCheckReference{{HealthCheck: hc.SelfLink}}
+				if err := gce.RemoveHealthChecksFromTargetPool(loadBalancerName, gce.region, healthchecks); err != nil {
+					return fmt.Errorf("failed to remove healthcheck from target pool for load balancer (%s): %v", lbRefStr, err)
+				}
+				if err := gce.deleteHealthCheckAndFirewall(svc, loadBalancerName, clusterID, hc.Name); err != nil {
+					return fmt.Errorf("failed to delete healthcheck for load balancer (%s): %v", lbRefStr, err)
+				}
+			}
+		}
+
+		if hcToCreate != nil {
+			if hcCreated, err := gce.createHealthCheckAndFirewall(svc, loadBalancerName, serviceName.String(), ipAddressToUse, gce.region, clusterID, hosts, hcToCreate); err != nil {
+				return err
+			} else {
+				healthchecks := []*compute.HealthCheckReference{{HealthCheck: hcCreated.SelfLink}}
+				if err := gce.AddHealthChecksToTargetPool(loadBalancerName, gce.region, healthchecks); err != nil {
```

Comment:
What happen if we pass in healthcheck that is already referred by the targetpool?

Reply:
Check added.